src/Elmish/HTML/Internal.purs

```diff
@@ -1,9 +1,90 @@
 module Elmish.HTML.Internal where
 
-import Elmish.React.Import (ImportedReactComponent)
+import Prelude
+
+import Elmish (ReactElement, createElement, createElement')
+import Elmish.Foreign (class CanPassToJavaScript)
+import Elmish.React (class ReactChildren, class ValidReactProps)
+import Elmish.React.Import (class IsSubsetOf, CommonProps, ImportedReactComponent)
+import Record (merge)
+import Type.Row (type (+))
+import Type.Row.Homogeneous (class Homogeneous)
 import Unsafe.Coerce (unsafeCoerce)
 
+-- | The type of `style` props in React elements. Construct values of this type
+-- | via the `css` function. For example:
+-- |
+-- |    div { style: css { height: "50px", width: "50px" } }
+-- |
 foreign import data CSS :: Type
+instance jsCSS :: CanPassToJavaScript CSS
+
+
+-- | Construct a value of type `CSS`, which is the type of `style` props, out of
+-- | a record with all `String` values. For example:
+-- |
+-- |    div { style: css { height: "50px", width: "50px" } }
+-- |
+-- | There is currently no type safety regarding the specific fields admitted in
+-- | the style, or different types of those fields. This has been deemed "good
+-- | enough" for now.
+css :: forall r. Homogeneous r String => { | r } -> CSS
```

Comment:
Hmm, I think React requires some `style` values to be numbers, e.g.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b20d3d2e1f0a46d755d5b7a936fa1bc76218e31b/types/react/v15/index.d.ts#L1196-L1200

Can we remove the `Homogenous` constraint since it doesn’t provide much safety anyway?

In the future, we could use [`csstype`](https://www.npmjs.com/package/csstype) to generate PureScript type definitions: https://fettblog.eu/typescript-react/styles/

Reply:
Sure